environments/common/src/main/java/org/jboss/weld/environment/deployment/discovery/jandex/JandexClassFileServices.java

```diff
@@ -74,10 +77,20 @@ public JandexClassFileServices(JandexDiscoveryStrategy strategy) {
         }
         this.classLoader = WeldResourceLoader.getClassLoader();
         this.annotationClassAnnotationsCache = ComputingCacheBuilder.newBuilder().build(new AnnotationClassAnnotationLoader());
+        try {
+            Method setFlags = ClassFileInfo.class.getMethod("setFlags");
```

Comment:
`ClassFileInfo` should be `org.jboss.jandex.ClassInfo`. Also use `org.jboss.weld.security.GetDeclaredMethodAction` instead of `Class.getMethod()` (which wouldn't work anyway because `ClassInfo.setFlags()` is not public).